src/pages/Profile.tsx

```diff
@@ -124,6 +128,59 @@ const Profile = () => {
   });
   const [editProfileModalOpen, setEditProfileModalOpen] = useState(false);
   const [loading, setLoading] = useState(true);
+  const [followListModalLoading, setFollowListModalLoading] = useState(true);
+  const [followListModelOpen, setFollowListModelOpen] = useState(false);
+  const [listType, setListType] = useState<ListType | null>(null);
+  const [followerListData, setFollowerListData] = useState<NetworkUsers[]>([]);
+
+  const onFollowed = (isFollowing: boolean) => {
+    setProfileContents((prevProfileContents) => ({
+      ...prevProfileContents,
+      followingCount: isFollowing
+        ? --prevProfileContents.followingCount
+        : ++prevProfileContents.followingCount,
+    }));
+  };
+
+  const handleOpenFollowersModal = async () => {
+    setListType("Followers");
```

Comment:
I'm not really a fan of this `ListType` thing we have going on here. Firstly, I don't like that it's a string, I think if we wanted to do types like this they should be enums

I'd prefer if we get rid of it entirely though, cause something about passing flags into components to tell them where to fetch data from seems wrong to me. Passing in the url endpoint also feels greasy too. My opinion is that the lowest level components shouldn't have to worry about the data; all they need to know is the _shape_ of the data being passed to them, and then they can render the UI appropriately

So for our case, I'm proposing we have 2 separate modals in here and conditionally render them based on what the user clicked. Since the endpoints to grab the follower and following data are completely different, we should fetch for them separately in the calling component and just pass them into a generic follower/following modal component. Then as for the modal title, we can make that into a prop

Reply:
Done, Ive updated the code to have two separate modals that are conditionally rendered and I changed the `listType` prop to be `title` instead.